pkg/controllers/placementdecision/creating_controller.go

```diff
@@ -0,0 +1,152 @@
+)
+
+// placementDecisionCreatingController creates PlacementDecisions for Placements
+type placementDecisionCreatingController struct {
+	clusterClient           clusterclient.Interface
+	placementLister         clusterlisterv1alpha1.PlacementLister
+	placementDecisionLister clusterlisterv1alpha1.PlacementDecisionLister
+}
+
+// NewPlacementDecisionCreatingController return an instance of placementDecisionCreatingController
+func NewPlacementDecisionCreatingController(
+	clusterClient clusterclient.Interface,
+	placementInformer clusterinformerv1alpha1.PlacementInformer,
+	placementDecisionInformer clusterinformerv1alpha1.PlacementDecisionInformer,
+	recorder events.Recorder,
+) factory.Controller {
+	c := placementDecisionCreatingController{
+		clusterClient:           clusterClient,
+		placementLister:         placementInformer.Lister(),
+		placementDecisionLister: placementDecisionInformer.Lister(),
+	}
+
+	return factory.New().
+		WithInformersQueueKeyFunc(func(obj runtime.Object) string {
+			key, _ := cache.MetaNamespaceKeyFunc(obj)
+			return key
+		}, placementInformer.Informer()).
+		WithFilteredEventsInformersQueueKeyFunc(func(obj runtime.Object) string {
+			accessor, _ := meta.Accessor(obj)
+			labels := accessor.GetLabels()
+			placementName := labels[placementLabel]
+			return fmt.Sprintf("%s/%s", accessor.GetNamespace(), placementName)
+		}, func(obj interface{}) bool {
+			accessor, err := meta.Accessor(obj)
+			if err != nil {
+				return false
+			}
+			labels := accessor.GetLabels()
+			if _, ok := labels[placementLabel]; ok {
+				return true
+			}
+			return false
+		}, placementDecisionInformer.Informer()).
+		WithSync(c.sync).
+		ToController("PlacementDecisionCreatingController", recorder)
+}
+
+func (c *placementDecisionCreatingController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	queueKey := syncCtx.QueueKey()
+	namespace, name, err := cache.SplitMetaNamespaceKey(queueKey)
+	if err != nil {
+		// ignore placement whose key is not in format: namespace/name
+		return nil
+	}
+
+	klog.V(4).Infof("Reconciling placement %q", queueKey)
+	placement, err := c.placementLister.Placements(namespace).Get(name)
+	if errors.IsNotFound(err) {
+		// no work if placement is deleted
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	// no work if placement is deleting
+	if !placement.DeletionTimestamp.IsZero() {
+		return nil
+	}
+
+	// query placementdecisions with label selector
+	requirement, err := labels.NewRequirement(placementLabel, selection.Equals, []string{placement.Name})
+	if err != nil {
+		return err
+	}
+	labelSelector := labels.NewSelector().Add(*requirement)
+	placementDecisions, err := c.placementDecisionLister.PlacementDecisions(namespace).List(labelSelector)
+	if err != nil {
+		return err
+	}
+
+	// no work if PlacementDecision has been created
+	if len(placementDecisions) > 0 {
+		return c.updateOwnerReferences(ctx, placement, placementDecisions)
```

Comment:
do we need to update owner here?

Reply:
Deleted the logic to update owner reference.